eth/chains/base.py

```diff
@@ -577,13 +585,23 @@ def import_block(self, block: BaseBlock, perform_validation: bool=True) -> BaseB
             ensure_imported_block_unchanged(imported_block, block)
             self.validate_block(imported_block)
 
-        self.chaindb.persist_block(imported_block)
+        new_canonical_headers, old_headers = self.chaindb.persist_block(imported_block)
         self.logger.debug(
             'IMPORTED_BLOCK: number %s | hash %s',
             imported_block.number,
             encode_hex(imported_block.hash),
         )
-        return imported_block
+
+        new_canonical_blocks = [
+            self.get_block_by_header(header) for header in new_canonical_headers]
+        old_canonical_blocks = [
+            self.get_block_by_header(header) for header in old_headers]
```

Comment:
You can drop the `tuple` casting for the return statement by just embedding the comprehension here in a `tuple(...)`

```python
squares = tuple(i ** 2 for i in range(10))
```